prusti-viper/src/encoder/encoder.rs

```diff
@@ -528,21 +415,23 @@ impl<'v, 'tcx> Encoder<'v, 'tcx> {
                     // one to pick.
                     let item = procs[0];
                     if let Some(spec) = self.get_procedure_specs(item.def_id) {
-                        impl_spec = spec.clone();
+                        impl_spec = spec;
                     } else {
                         debug!("Procedure {:?} has no specification", item.def_id);
                     }
+                } else {
+                    panic!("multiple trait method declarations found");
```

Comment:
I added this `panic` to see whether some of the failing tests are caused by this. It caused more tests to fail though. I have not investigated why we get multiple method yet. Ideas?

Reply:
Nothing comes to my mind at the moment.

Reply:
This panic is executed even when `procs.len() == 0`. Is it intentional? It doesn't match the panic message.

Reply:
@fpoli You are right, about half of the cases are triggered with `procs.len() == 0`. I guess these cases should be ok, since we just don't have any spec to find. But if multiple methods are found, we risk selecting the wrong one or ignoring an existing spec. Anyway, for the time being I am disabling this panic, since this is not really part of this PR…